### ISSUE TITLE
[Patch] Support for printing SocketCAN formated frames

### ISSUE THREAD

**Issue by guyharris** [the-tcpdump-group/tcpdump#253]
Converted from [SourceForge issue 2876645](http://sourceforge.net/support/tracker.php?aid=2876645), submitted by asu_fo

Libpcap gets extended to capture CAN frames via the SocketCAN interface.

This patch enabled tcpdump to print that captured frames for testing purposes and convenience.  The printing routine is taken from the can-utlis of SocketCAN.

**Comment by Adraub** [comment]
+1!
Does it just output the same thing as the "candump" socketCAN library?
I was disappointed when i tried to pipe through SSH SocketCAN frames (just the same as you can do with Ethernet frames), and tcpdump told me it wasn't possible.
It is really useful when you try to debug embedded systems.
I scripted something that works with python but i can help to migrate it :)
It could be a great feature for a future version.

**Comment by infrastation** [comment]
@Adraub, do you have a packet capture of this type?

**Comment by Adraub** [comment]
On ubuntu using socketcan you can easily create a capture @infrastation 

 $ modprobe vcan
 $ sudo ip link add dev vcan0 type vcan
 $ sudo ip link set up vcan0

let this run in a terminal:
cangen vcan0

in another launch:
tcpdump -s0 -U -n -w - -i vcan0 | wireshark -k -i -

Here you are, you can save your can file as pcap.

Please note that sadly, i see on my device can frame send twice, one as a "kernel" request, and another as a received frame. This is linux cooked capture and it doesnt seem possible to filter it directly with tcpdump.

**Comment by guyharris** [comment]
> it's possible using the -w flag

Yes, for *all* link-layer header types, even Ethernet, if you want to run tcpdump and pipe its output to Wireshark, you need `-w`.
